internal/core/src/bitset/detail/platform/x86/avx2-impl.h

```diff
@@ -1589,26 +1606,67 @@ OpArithCompareImpl<float, AOp, CmpOp>::op_arith_compare(
     if constexpr (AOp == ArithOpType::Mod) {
         return false;
     } else {
-        // the restriction of the API
-        assert((size % 8) == 0);
+        if constexpr (AOp == ArithOpType::Div) {
+            if (std::isfinite(value) && std::isfinite(right_operand) &&
+                right_operand > 0) {
+                // a special case that allows faster processing by using the multiplication
+                //   operation instead of the division one.
+
+                // the restriction of the API
+                assert((size % 8) == 0);
+
+                //
+                const __m256 right_v = _mm256_set1_ps(right_operand);
+                const __m256 value_v = _mm256_set1_ps(value);
+
+                // todo: aligned reads & writes
+
+                const size_t size8 = (size / 8) * 8;
+                for (size_t i = 0; i < size8; i += 8) {
+                    const __m256 v0s = _mm256_loadu_ps(src + i);
+                    const __m256 cmp = ArithHelperF32<AOp, CmpOp>::op_special(
+                        v0s, right_v, value_v);
+                    const uint8_t mmask = _mm256_movemask_ps(cmp);
+
+                    res_u8[i / 8] = mmask;
+                }
+
+                return true;
+            } else if (std::isfinite(value) && std::isfinite(right_operand) &&
```

Comment:
these codes are duplicated with the default case. can we just check `std::isfinite(value) && std::isfinite(right_operand) && right_operand < 0` here?